sycl/include/sycl/detail/common.hpp

```diff
@@ -379,6 +384,43 @@ static constexpr std::array<T, N> RepeatValue(const T &Arg) {
 // classes.
 struct AllowCTADTag;
 
+// Look up a function name that was dynamically linked
+// This is used by the runtime where it needs to manipulate native handles (e.g.
+// retaining OpenCL handles). On Windows, the symbol name is looked up in
+// `WinName`. In Linux, it uses `LinName`.
+//
+// The library must already have been loaded (perhaps by UR), otherwise this
+// function throws.
+template <typename fn>
+fn *dynLookupFunction([[maybe_unused]] const char *WinName,
```

Comment:
Is there a more suitable place for this than common.hpp?

Reply:
Probably it should go into `os_util.hpp` and/or `os_util.cpp`  There are also `windows_os_utils.hpp` and friends in `sycl/source/detail/`